src/Recursor.h

```diff
@@ -611,17 +611,17 @@ void Recursor<Derived>::ExtendBeta(const M& beta, size_t lastColumn, M& ext, int
         std::tie(beginRow, endRow) = beta.UsedRowRange(lastColumn + 1);
     else
         std::tie(beginRow, endRow) = beta.UsedRowRange(lastColumn);
-    for (int j = 0; j <= lastColumn && j <= numExtColumns; ++j)
+    for (size_t j = 0; j <= lastColumn && j <= numExtColumns; ++j)
         beginRow = std::min(static_cast<int>(beta.UsedRowRange(lastColumn - j).first), beginRow);
 
-    for (int j = lastColumn; j + numExtColumns > lastColumn; j--) {
+    for (int j = lastColumn; j + numExtColumns - lastColumn > 0; j--) {
         /* Convert from old template to new template coordinates.
            lengthDiff will be 0 for substitution, -1 for deletion and +1 for
            insertion
          */
-        int jp = j + lengthDiff;
+        size_t jp = j + lengthDiff;
         // What is the current extension column we are adding data into.
-        int extCol = lastExtColumn - (lastColumn - j);
+        size_t extCol = lastExtColumn - (lastColumn - j);
```

Comment:
This math seems correct, please delete the previous draft comments (but add here the note that j+lengthDiff>=0 was established above).

I agree about jp, but let's consider that separate from this PR.  It may also be worth adding comments or even assertions regarding the identities that hold among the variables.